### PR TITLE
add generator modification

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**
Signed-off-by: jacques borsenberger <jacques.borsenberger@rte-france.com>

### CODE REVIEW

File: src/components/dialogs/generator-modification-dialog.js [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+                fullWidth
+                maxWidth="md" // 3 columns
+                open={open}
+                onClose={handleClose}
+                aria-labelledby="dialog-create-generator"
+            >
+                <DialogTitle>
+                    <Grid container justifyContent={'space-between'}>
+                        <Grid item xs={11}>
+                            {/* TODO message */}
```
Comment (niconoir) [inline src/components/dialogs/generator-modification-dialog.js:314]: Can be removed ?

**Review by niconoir — CHANGES_REQUESTED** [review]

**Review by niconoir — APPROVED** [review]

File: src/components/dialogs/generator-modification-dialog.js (hunk 1 above)
Comment (JacquesBorsenberger) [reply src/components/dialogs/generator-modification-dialog.js]: ddone

File: src/components/dialogs/generator-modification-dialog.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        validation: {
+            isFieldNumeric: true,
+            isValueGreaterThan: '0',
+            errorMsgId: 'VoltageGreaterThanZero',
+        },
+        adornment: VoltageAdornment,
+        formProps: { disabled: voltageRegulation === true },
+        inputForm: inputForm,
+        defaultValue: getValue(formValues?.voltageSetpoint),
+        previousValue: generatorInfos?.targetP,
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/dialogs/generator-modification-dialog.js]: done

File: src/components/dialogs/generator-modification-dialog.js [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+    const [activePowerSetpoint, activePowerSetpointField] = useDoubleValue({
+        label: 'ActivePowerText',
+        validation: {
+            isFieldNumeric: true,
+        },
+        adornment: ActivePowerAdornment,
+        inputForm: inputForm,
+        defaultValue: getValue(formValues?.activePowerSetpoint),
+        previousValue: generatorInfos?.p,
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/dialogs/generator-modification-dialog.js]: done

**Pull request merged**